test/auth/signup/sign_up_test.dart

```diff
@@ -0,0 +1,74 @@
+
+import 'package:chat_app_front/auth/signup/sign_up_form.dart';
+import 'package:chat_app_front/auth/signup/sign_up_keys.dart';
+import 'package:chat_app_front/auth/signup/sign_up_strings.dart';
+import 'package:chat_app_front/auth/signup/sign_up_user.dart';
+import 'package:chat_app_front/auth/user_repository.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+class MockSignUpUser extends Mock implements SignUpUser{}
+
+class MockNavigatorObserver extends Mock implements NavigatorObserver{}
+
+
+main(){
+  SignUpUser mockSignUpUser;
+  NavigatorObserver mockObserver;
+  setUp((){
+    mockObserver = MockNavigatorObserver();
+    mockSignUpUser = MockSignUpUser();
+  });
+
+  Future<Null> _buildSignUpPage(WidgetTester tester) async{
+    await tester.pumpWidget(MaterialApp(
+      home:Scaffold(
+        appBar: AppBar(title: Text(SignUpStrings.appBarText),),
+        body: Container(
+          child: ListView(
+            children: <Widget>[
+              Column(
+                children: <Widget>[
+                  SignUpForm(signUpUser: mockSignUpUser,)
+                ],
+              ),
+            ],
+          ),
+        ),
+      ),
+      navigatorObservers: [mockObserver],
+    ));
+    verify(mockObserver.didPush(any, any));
+  }
+
+  final username1 = "test1";
```

Comment:
Czemu to jest wewnątrz main?

Reply:
Dodatkowo, sugeruję precyzyjniejsze nazwy, np. validUserName, validEmail, invalidEmail, validPassword, tooShortPassword itp. itd.